src/connectors/queue/payTo/payToByBlockchain.ts

```diff
@@ -0,0 +1,364 @@
+    } catch (error) {
+      await trx.rollback()
+      throw error
+    }
+  }
+
+  private failBothTxAndBlockchainTx = async (
+    txId: string,
+    blockchainTxId: string
+  ) => {
+    await this.updateTxAndBlockchainTxState(
+      { txId, txState: TRANSACTION_STATE.failed },
+      {
+        blockchainTxId,
+        blockchainTxState: BLOCKCHAIN_TRANSACTION_STATE.reverted,
+      }
+    )
+  }
+
+  private timeoutBlockchainTx = async (blockchainTxId: string) => {
+    await this.paymentService.markBlockchainTransactionStateAs({
+      id: blockchainTxId,
+      state: BLOCKCHAIN_TRANSACTION_STATE.timeout,
+    })
+  }
+
+  private validateTxLogs = async (
+    logs: Log[],
+    {
+      curatorAddress,
+      creatorAddress,
+      cid,
+      tokenAddress,
+      amount,
+      decimals,
+    }: {
+      curatorAddress?: string
+      creatorAddress?: string
+      cid: string
+      tokenAddress: string
+      amount: string
+      decimals: number
+    }
+  ) => {
+    const abi = [
+      'event Curation(address indexed curator, address indexed creator, address indexed token, string uri, uint256 amount)',
+    ]
+    const topic =
+      '0xc2e41b3d49bbccbac6ceb142bad6119608adf4f1ee1ca5cc6fc332e0ca2fc602'
+    if (logs.length === 0) {
+      return false
+    } else {
+      for (const log of logs) {
+        if (
+          log.address.toLowerCase() ===
+            environment.curationContractAddress.toLowerCase() &&
+          log.topics[0] === topic
+        ) {
+          const iface = new ethers.utils.Interface(abi)
+          const event = iface.parseLog(log)
+          if (
+            event.args.curator!.toLowerCase() ===
+              curatorAddress!.toLowerCase() &&
+            event.args.creator!.toLowerCase() ===
+              creatorAddress!.toLowerCase() &&
+            event.args.token!.toLowerCase() === tokenAddress.toLowerCase() &&
+            event.args.amount!.toString() ===
+              toTokenBaseUnit(amount, decimals) &&
+            (event.args.uri!.includes(cid) ||
+              event.args.uri!.includes(tryConvertVersionOfCID(cid)))
+          ) {
+            return true
+          }
+        }
+      }
+    }
+    return false
+  }
+
+  /**
+   * Pay-to handler.
+   *
+   */
+  private handlePayTo: Queue.ProcessCallbackFunction<unknown> = async (job) => {
+    const data = job.data as PaymentParams
+    const txId = data.txId
+
+    const tx = await this.paymentService.baseFindById(txId)
+    if (!tx) {
+      throw new PaymentQueueJobDataError('pay-to pending tx not found')
+    }
+
+    if (tx.provider !== PAYMENT_PROVIDER.blockchain) {
+      throw new PaymentQueueJobDataError('wrong pay-to queue')
+    }
+
+    const blockchainTx =
+      await this.paymentService.findBlockchainTransactionById(tx.providerTxId)
+
+    if (!blockchainTx) {
+      throw new PaymentQueueJobDataError('blockchain transaction not found')
+    }
+
+    const provider = getProvider()
+    const confirms = 1
```

Comment:
128 is a safe number to avoid reorg for Polygon?

Reply:
the queue job is for updating transaction  state timely